staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1367,28 +1368,39 @@ type Condition struct {
 	// Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be
 	// useful (see .node.status.conditions), the ability to deconflict is important.
 	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Available;Progressing;Degraded
 	Type string `json:"type" protobuf:"bytes,1,opt,name=type"`
 	// status of the condition, one of True, False, Unknown.
 	// +required
+	// +kubebuilder:validation:Required
 	Status ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status"`
 	// observedGeneration represents the .metadata.generation that the condition was set based upon.
 	// For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date
 	// with respect to the current state of the instance.
 	// +optional
+	// +kubebuilder:validation:Minimum=0
 	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,3,opt,name=observedGeneration"`
 	// lastTransitionTime is the last time the condition transitioned from one status to another.
 	// This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.
 	// +required
+	// +kubebuilder:validation:Required
```

Comment:
is there a time format available?

Reply:
it looks like we can use [type: string and format: date-time](https://book.kubebuilder.io/reference/markers/crd-validation.html)